File: src/server/src/model/entities/episode.ts
import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, ManyToOne, Unique, OneToMany } from "typeorm"
import { Show } from "./show"
import { Chunk } from "./chunk"

@Entity()
@Unique(['show', 'season', 'episode'])
export class Episode extends BaseEntity {
    @PrimaryGeneratedColumn()
    id!: number

    @ManyToOne(() => Show, show => show.episodes, { onDelete: "CASCADE"})
    show!: Show

    @OneToMany(() => Chunk, chunk => chunk.episode, { cascade: true })
    chunks!: Chunk[]

    @Column()
    season!: number

    @Column()
    episode!: number

    @Column()
    title!: string
}

File: src/client/lib/features/subscriptions/models/episode.dart
class Episode {
  final int id;
  final int season;
  final int number;
  final String title;

  Episode({
    required this.id,
    required this.season,
    required this.number,
    required this.title,
  });

  Map<String, dynamic> toMap() {
    return {'id': id, 'season': season, 'episode': number, 'title': title};
  }

  factory Episode.fromMap(Map<String, dynamic> map) {
    return Episode(
      id: map['id'] ?? 0,
      season: map['season'] ?? 0,
      number: map['episode'] ?? 0,
      title: map['title'] ?? '',
    );
  }
}
